src/roman_datamodels/validate.py

```diff
@@ -1,191 +0,0 @@
-    """
-    var = os.getenv(ROMAN_VALIDATE)
-
-    if not (validate := var is None or var.lower() in ["true", "yes", "1"]):
-        validation_is_disabled()
-
-    return validate
-
-
-def strict_validation_is_disabled():
-    MESSAGE = dedent(
-        """\
-            Strict validation has been disabled. This may result
-            in validation errors arising when writing data.
-
-            To turn strict validation back on, unset the
-            environment variable ROMAN_STRICT_VALIDATION or set
-            it to "true".
-        """
-    )
-
-    warnings.warn(MESSAGE, ValidationWarning)
-
-
-def will_strict_validate():
-    """
-    Determine if strict validation is enabled.
-    """
-    var = os.getenv(ROMAN_STRICT_VALIDATION)
-
-    if not (validate := var is None or var.lower() in ["true", "yes", "1"]):
-        strict_validation_is_disabled()
-
-    return validate
-
-
-class ValidationWarning(Warning):
-    pass
-
-
-def value_change(value, pass_invalid_values, strict_validation):
-    """
-    Validate a change in value against a schema.
-    Trap error and return a flag.
-    """
-    try:
-        _check_value(value)
-        update = True
-    except ValidationError as errmsg:
-        update = False
-        if pass_invalid_values:
-            update = True
-        if strict_validation:
-            raise errmsg
-        else:
-            warnings.warn(errmsg, ValidationWarning)
-    return update
-
-
-def _check_type(validator, types, instance, schema):
-    """
-    Callback to check data type. Skips over null values.
-    """
-    if instance is None:
-        errors = []
-    else:
-        errors = asdf_schema.validate_type(validator, types, instance, schema)
-    return errors
-
-
-validator_callbacks = HashableDict(asdf_schema.YAML_VALIDATORS)
-validator_callbacks.update({"type": _check_type})
-
-
-def _check_value(value):
-    """
-    Perform the actual validation.
-    """
-
-    validator_context = AsdfFile()
-
-    if hasattr(value, "_schema"):
-        temp_schema = value._schema()
-    else:
-        temp_schema = {"$schema": "http://stsci.edu/schemas/asdf-schema/0.1.0/asdf-schema"}
-    validator = asdf_schema.get_validator(temp_schema, validator_context, validators=validator_callbacks)
-
-    value = yamlutil.custom_tree_to_tagged_tree(value, validator_context)
-    validator.validate(value, _schema=temp_schema)
-    validator_context.close()
-
-
-def _error_message(path, error):
-    """
-    Add the path to the attribute as context for a validation error
-    """
-    if isinstance(path, list):
-        spath = [str(p) for p in path]
-        name = ".".join(spath)
-    else:
-        name = str(path)
-
-    error = str(error)
-    if len(error) > 2000:
-        error = error[0:1996] + " ..."
-    errfmt = "While validating {} the following error occurred:\n{}"
-    errmsg = errfmt.format(name, error)
-    return errmsg
-
-
-@contextmanager
-def nuke_validation():
```

Comment:
I'm not sure we want to remove this feature yet. It was made so that IPAC and other external devs could have a code independent method (environment variable) to turn off validation.

Yes, this PR removes the validation on assignment but `nuke_validation` also turns off validation on both file open and file write.

Reply:
Thanks for taking a look. With this PR disabling [validation_on_read](https://asdf.readthedocs.io/en/latest/asdf/config.html#validate-on-read) allows opening invalid files.

This leaves only writing invalid files (which is possible with `nuke_validation`). Would you provide some more details about when saving a knowingly invalid file is useful? I'd like to better understand why this is useful.

Reply:
Unfortunately, the `validation_on_read` option does satisfy the request that `nuke_validation` was created for. In that, they wanted a method to disable validation on a code base without having to make changes to that code base itself. `validation_on_read` requires one to actively change an argument within the code to ASDF in order to use while `nuke_validation` allows for one to disable setting an environment variable on their system.

As to why they wanted the ability to write invalid files? That is because they were starting with invalid files and simply wanted to save the results of their code into a new version of the input file. They did not want to deal with fixing whatever was wrong to cause the input file to be invalid just to save what they cared about.

The original motivation for the request was constant expressed frustration with "ASDF" always throwing "useless" validation errors. The real issue was that RAD is/was under fairly constant flux meaning that their input files were constantly becoming invalid for for reasons that were irrelevant to what they were doing and the constant updating and distribution of new "valid" input files was causing a lot of issues. This maybe less of an issue now that RAD is more "stable", but I think it should remain until we "release" a final version of all the RAD schemas (any changes to RAD schemas result in a new schema with a version bump to the given schema).